atc/lidar/scanner.go

```diff
@@ -46,6 +46,17 @@ func (s *scanner) Run(ctx context.Context) error {
 	waitGroup := new(sync.WaitGroup)
 	resourceTypesChecked := &sync.Map{}
 
+	s.scanResources(spanCtx, resources, resourceTypes, waitGroup, resourceTypesChecked)
+	waitGroup.Wait()
+	s.scanUncheckedResourceTypes(spanCtx, resourceTypes, waitGroup, resourceTypesChecked)
```

Comment:
Now that we check all resource types, does `s.check` still need to check the parent resource type? Or perhaps the order the checks are created in matters

Reply:
I think it still makes sense for `s.check` to check the parent resource-type first before running the resource's check. It feels out of order to run a check for a resource and then check if it's base resource-type has updated. I'm going to leave this as-is.